superset-frontend/src/explore/exploreUtils.js

```diff
@@ -309,19 +310,28 @@ export const useDebouncedEffect = (effect, delay) => {
   }, [callback, delay]);
 };
 
-export const getSimpleSQLExpression = (
-  subject,
-  operator,
-  comparator,
-  isMulti,
-) => {
+export const getSimpleSQLExpression = (subject, operator, comparator) => {
+  const isMulti = MULTI_OPERATORS.has(operator);
   let expression = subject ?? '';
   if (subject && operator) {
     expression += ` ${operator}`;
-    if (comparator) {
-      expression += ` ${isMulti ? "('" : ''}${comparator}${
-        isMulti ? "')" : ''
-      }`;
+    const firstValue =
+      isMulti && Array.isArray(comparator) ? comparator[0] : comparator;
+    const comparatorArray =
+      comparator === undefined ||
+      comparator === null ||
+      Array.isArray(comparator)
+        ? comparator
+        : [comparator];
+    const isString =
+      firstValue !== undefined && Number.isNaN(Number(firstValue));
+    const quote = isString ? "'" : '';
+    const [prefix, suffix] = isMulti ? ['(', ')'] : ['', ''];
+    const formattedComparators = (comparatorArray || []).map(
+      val => `${quote}${isString ? val.replace("'", "''") : val}${quote}`,
+    );
+    if (comparatorArray) {
+      expression += ` ${prefix}${formattedComparators.join(', ')}${suffix}`;
```

Comment:
This isn't super easy to read (apologies), but the unit tests should be easier to reason about.